packages/shorebird_cli/lib/src/cache.dart

```diff
@@ -171,6 +178,25 @@ allowed to access $storageUrl.''',
 
     await extractArtifact(response.stream, location.path);
 
+    final expectedChecksum = checksum;
+    if (expectedChecksum != null) {
+      final artifactFile = File(p.join(location.path, name));
+
+      if (!checksumChecker.checkFile(artifactFile, expectedChecksum)) {
+        // Delete the location, so if the download is retried, it will be
+        // re-downloaded.
+        location.deleteSync(recursive: true);
+        // TODO(erickzanardo): Automatically retry the download.
```

Comment:
Thought on doing this in a follow up for easier to review PRs

Reply:
You probably want to do the retry at a higher level (the thing that call this), but yeah.